### PR TITLE
Add metadata map to Example

### PR DESCRIPTION
**Pull request by TDDFT**
Metadata are not features but these values can help during training/evaluation.

@deerzq @jq @timyitong

### CODE REVIEW

File: core/src/main/java/com/airbnb/aerosolve/core/features/Features.java [hunk 1]
```diff
@@ -73,31 +73,43 @@ public Example toExample(boolean isMultiClass) {
...
-        } else {
-          addNumberFeature((Number) value, feature, floatFeatures);
         }
       }
     }
     return example;
   }
 
+  private static boolean isMetadata(String name) {
+    return name.charAt(0) == FAMILY_SEPARATOR;
```
Comment (deerzq) [inline core/src/main/java/com/airbnb/aerosolve/core/features/Features.java]: probably better to define a new special sign for meta data

Comment (TDDFT) [inline core/src/main/java/com/airbnb/aerosolve/core/features/Features.java]: `#`? Not sure whether this or other characters are good choices for Hive column names though.

Comment (deerzq) [inline core/src/main/java/com/airbnb/aerosolve/core/features/Features.java]: e.g. `_meta` ? just want to be more specific?

**Comment by TDDFT** [comment]
@deerzq Changed `_` to `_meta_` accordingly.

**Comment by deerzq** [comment]
LGTM

**Comment by TDDFT** [comment]
I guess you should click the merge button. I am not authorized to merge to aerosolve yet.

**Pull request merged**